include/irr/asset/IVirtualTexture.h

```diff
@@ -527,71 +625,68 @@ class IVirtualTexture : public core::IReferenceCounted
-            std::to_string(_set) + "/" +
-            std::to_string(_pgtBinding) + "/" +
-            std::to_string(_fsamplersBinding) + "/" +
-            std::to_string(_isamplersBinding) + "/" +
-            std::to_string(_usamplersBinding) + "/" +
-            (m_fsamplers.views ? std::to_string(m_fsamplers.views->size()) : "0") + "/" +
-            (m_isamplers.views ? std::to_string(m_isamplers.views->size()) : "0") + "/" +
-            (m_usamplers.views ? std::to_string(m_usamplers.views->size()) : "0");
+        return "irr/builtin/glsl/virtual_texturing/descriptors.glsl";
     }
-    std::string getGLSLFunctionsIncludePath(const std::string& _get_pgtab_sz_log2_name, const std::string& _get_phys_pg_tex_sz_rcp_name, const std::string& _get_vtex_sz_rcp_name, const std::string& _get_layer2pid) const
+    std::string getGLSLFunctionsIncludePath() const
     {
         //functions.glsl/pg_sz_log2/tile_padding/pgtab_tex_name/phys_pg_tex_name/get_pgtab_sz_log2_name/get_phys_pg_tex_sz_rcp_name/get_vtex_sz_rcp_name/get_layer2pid/(addr_x_bits/addr_y_bits)...
         std::string s = "irr/builtin/glsl/virtual_texturing/functions.glsl/";
         s += std::to_string(m_pgSzxy_log2) + "/";
-        s += std::to_string(m_tilePadding) + "/";
-        s += _get_pgtab_sz_log2_name + "/";
-        s += _get_phys_pg_tex_sz_rcp_name + "/";
-        s += _get_vtex_sz_rcp_name + "/";
-        s += _get_layer2pid;
+        s += std::to_string(m_tilePadding);
 
         return s;
     }
 
 protected:
+
+
     template <typename DSlayout_t>
-    core::smart_refctd_dynamic_array<typename DSlayout_t::SBinding> getDSlayoutBindings_internal(uint32_t _pgtBinding = 0u, uint32_t _fsamplersBinding = 1u, uint32_t _isamplersBinding = 2u, uint32_t _usamplersBinding = 3u) const
+    auto getDSlayoutBindings_internal(uint32_t _pgtBinding = 0u, uint32_t _fsamplersBinding = 1u, uint32_t _isamplersBinding = 2u, uint32_t _usamplersBinding = 3u) const
     {
-        using retval_t = core::smart_refctd_dynamic_array<typename DSlayout_t::SBinding>;
-        auto retval = core::make_refctd_dynamic_array<retval_t>(1u+(getFloatViews().size()?1u:0u)+(getIntViews().size()?1u:0u)+(getUintViews().size()?1u:0u));
-        auto* bindings = retval->data();
+        using bindings_t = core::smart_refctd_dynamic_array<typename DSlayout_t::SBinding>;
+        using samplers_t = core::smart_refctd_dynamic_array<core::smart_refctd_ptr<sampler_t>>;
+        using retval_t = std::pair<bindings_t, samplers_t>;
+
+        auto bindings_array = core::make_refctd_dynamic_array<bindings_t>(1u+(getFloatViews().size()?1u:0u)+(getIntViews().size()?1u:0u)+(getUintViews().size()?1u:0u));
+        auto* bindings = bindings_array->data();
+        auto samplers_array = core::make_refctd_dynamic_array<samplers_t>(1u+std::max(getFloatViews().size(), std::max(getIntViews().size(), getUintViews().size())));
+        auto* samplers = samplers_array->data();
```

Comment:
why do you need so many samplers?

Reply:
for binding (as u can see)

Reply:
yeah, but you have one sampler per binding, whereas you probably need only 2 samplers, one for pagetable and integer textures and one for float textures

you don't need a unique sampler per binding, right?

Reply:
there are 2 samplers objects, but i need array (of length = `binding.count` = view count) of samplers for views binding

Reply:
ok I didn't see that there are only 2